src/mpid/ch4/netmod/ofi/ofi_rma.c

```diff
@@ -386,6 +388,7 @@ int MPIDI_OFI_pack_put(const void *origin_addr, int origin_count,
     req->noncontig.put.origin.datatype = origin_datatype;
     req->noncontig.put.origin.pack_offset = 0;
     req->noncontig.put.origin.total_bytes = origin_bytes;
+    MPIR_Datatype_add_ref_if_not_builtin(origin_datatype);
```

Comment:
Can you move this line directly beneath the line saving the datatype handle? I think they logically go together.